### PR TITLE
Rework generate account flow

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[sc-9422]

File: packages/common/src/store/accounts.slice.ts [hunk 1]
```diff
@@ -103,7 +104,10 @@ const slice = createSlice({
       state,
       action: PayloadAction<{ mnemonicPhrase: string; address: TAddress } | undefined>
     ) {
-      state.generatedAccount = action.payload;
+      state.generatedAccount = action.payload ? { ...action.payload, persistent: true } : undefined;
+    },
+    setGeneratedAccountPersistent(state, action: PayloadAction<boolean>) {
+      state.generatedAccount = { ...state.generatedAccount, persistent: action.payload };
```
Comment (Mrtenz) [inline packages/common/src/store/accounts.slice.ts:110]: Test please

File: packages/common/src/store/accounts.slice.ts [hunk 2]
```diff
@@ -120,11 +124,12 @@ export const {
...
   fetchFailed,
   fetchReset,
-  setGeneratedAccount
+  setGeneratedAccount,
+  setGeneratedAccountPersistent
 } = slice.actions;
 
 export const addSavedAccounts = createAction<boolean>(`${sliceName}/addSavedAccounts`);
-export const addGeneratedAccount = createAction<boolean>(`${sliceName}/addGeneratedAccount`);
+export const addGeneratedAccount = createAction(`${sliceName}/addGeneratedAccount`);
```
Comment (Mrtenz) [inline packages/common/src/store/accounts.slice.ts:132]: Can you specify a type

File: packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+describe('GenerateAccountAddress', () => {
+  it('renders', () => {
+    const { getByText } = getComponent(
+      createMockStore({
+        accounts: {
+          generatedAccount: {
+            mnemonicPhrase: 'test test test test test test test test test test test ball',
+            address: '0xc6D5a3c98EC9073B54FA0969957Bd582e8D874bf'
```
Comment (Mrtenz) [inline packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx:44]: Isn't there a fixture for this

File: packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          }
+        }
+      })
+    );
+    expect(getByText(translateRaw('NEW_ACCOUNT_TITLE'))).toBeDefined();
+  });
+
+  it('shows the address and mnemonic phrase', () => {
+    const mnemonicPhrase = 'test test test test test test test test test test test ball';
+    const address = '0xc6D5a3c98EC9073B54FA0969957Bd582e8D874bf';
```
Comment (Mrtenz) [inline packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx:54]: And this

File: packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  });
+
+  it('dispatches addGeneratedAccount when clicking the button', () => {
+    const onNext = jest.fn();
+    const { getByText } = getComponent(
+      createMockStore({
+        accounts: {
+          generatedAccount: {
+            mnemonicPhrase: 'test test test test test test test test test test test ball',
+            address: '0xc6D5a3c98EC9073B54FA0969957Bd582e8D874bf'
```
Comment (Mrtenz) [inline packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx:82]: And this

File: packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+  const handleToggle = () => dispatch(setGeneratedAccountPersistent(!persistent));
+  const handleClick = () => setShowMnemonicPhrase(!showMnemonicPhrase);
+
+  return (
+    <>
+      <Container>
+        {flowHeader}
+        <Box sx={{ textAlign: 'center' }} mb="3">
+          <Heading fontSize="24px" lineHeight="150%">
```
Comment (Mrtenz) [inline packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.tsx:24]: Is there a theme value for this? And other font sizes in this file

File: packages/quill/src/crypto/accounts.slice.ts [hunk 7]
```diff
@@ -63,7 +63,10 @@ const slice = createSlice({
       state,
       action: PayloadAction<{ mnemonicPhrase: string; address: TAddress } | undefined>
     ) {
-      state.generatedAccount = action.payload;
+      state.generatedAccount = action.payload ? { ...action.payload, persistent: true } : undefined;
+    },
+    setGeneratedAccountPersistent(state, action: PayloadAction<boolean>) {
+      state.generatedAccount = { ...state.generatedAccount, persistent: action.payload };
```
Comment (Mrtenz) [inline packages/quill/src/crypto/accounts.slice.ts:69]: Test please

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/common/src/store/accounts.slice.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/common/src/store/accounts.slice.ts:132]: What do you mean? There is no payload for this action anymore

File: packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx (hunk 3 above)
Comment (FrederikBolding) [reply packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.test.tsx:44]: There isn't but I can add one

File: packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.tsx (hunk 6 above)
Comment (FrederikBolding) [reply packages/quill/src/app/screens/GenerateAccount/GenerateAccountAddress.tsx:24]: No, it seems to be between two values. We do it this way everywhere else in the app.

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**